src/main/java/com/ea/restaurant/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.ea.restaurant.service.impl;
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import com.ea.restaurant.repository.UserRepository;
+import com.ea.restaurant.service.UserService;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+  private final UserRepository userRepository;
+
+  @Autowired
+  public UserServiceImpl(UserRepository userRepository) {
+    this.userRepository = userRepository;
+  }
+
+  @Override
+  public User save(User user) {
+    return userRepository.save(user);
+  }
+
+  @Override
+  public List<User> findAllEnabled() {
+    return userRepository.findAllByEntityStatus(Status.ACTIVE);
+  }
+
+  @Override
+  public Optional<User> findEnabledById(Long id) {
+    return userRepository.findByIdAndEntityStatus(id, Status.ACTIVE);
+  }
+
+  @Override
+  public void deleteUserById(Long id, User user) {
+    this.updateById(id, user);
+  }
+
+  @Override
+  public User updateById(Long id, User user) {
+    Optional<User> userToUpdate = this.findEnabledById(id);
```

Comment:
Please unwrap the user from the optional here, throw an exception for example if the user is not found

Reply:
Done.